kvserver: reproduce and fix range-size backpressure preventing spanconfig updates

dodeca12 opened the pull request:
This PR ensures that spanconfig updates involving garbage collection TTL changes or protected timestamp deletions can bypass the backpressure mechanism.

#### Implementation

**Commit 1: Reproduce the Issue**
- Added comprehensive test `TestSpanConfigUpdatesBlockedByRangeSizeBackpressureOnDefaultRangesWithKVAccessor` that reproduces the catch-22 scenario
- The test repeatedly writes spanconfig updates for a single key until the range becomes too large to split, triggering backpressure
- Demonstrates how spanconfig updates (including protected timestamp deletions) get blocked by backpressure:

```log
W250910 14:26:03.302675 144058 kv/kvserver/replica_backpressure.go:263  [T1,Vsystem,n1,s1,r77/1:/Table/4{7/1/"\xf…-8}] 481  applying backpressure to limit range growth on batch Put [/Table/47/1/"\xfa"/0], [txn: cfbe5191], [can-forward-ts]
I250910 14:26:03.304650 50 kv/kvserver_test/client_spanconfig_backpressure_test.go:209  [-] 482  ERROR! BREAKING OUT OF LOOP, numWrites successful: 62690, error: upsert-span-cfgs: split failed while applying backpressure to Put [/Table/47/1/"\xfa"/0], [txn: cfbe5191], [can-forward-ts] on range r77:/Table/4{7/1/"\xfa"-8} [(n1,s1):1, next=2, gen=1]: could not find valid split key
I250910 14:26:03.304650 50 kv/kvserver_test/client_spanconfig_backpressure_test.go:209  [-] 482 +(1) attached stack trace
I250910 14:26:03.304650 50 kv/kvserver_test/client_spanconfig_backpressure_test.go:209  [-] 482 +  -- stack trace:
I250910 14:26:03.304650 50 kv/kvserver_test/client_spanconfig_backpressure_test.go:209  [-] 482 +  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).execInternal.func1.1
I250910 14:26:03.304650 50 kv/kvserver_test/client_spanconfig_backpressure_test.go:209  [-] 482 +  | 	pkg/sql/internal.go:1260
I250910 14:26:03.304650 50 kv/kvserver_test/client_spanconfig_backpressure_test.go:209  [-] 482 +  | github.com/cockroachdb/cockroach/pkg/sql.(*rowsIterator).Close
I250910 14:26:03.304650 50 kv/kvserver_test/client_spanconfig_backpressure_test.go:209  [-] 482 +  | 	pkg/sql/internal.go:608
I250910 14:26:03.304650 50 kv/kvserver_test/client_spanconfig_backpressure_test.go:209  [-] 482 +  | github.com/cockroachdb/cockroach/pkg/sql.(*rowsIterator).Next
I250910 14:26:03.304650 50 kv/kvserver_test/client_spanconfig_backpressure_test.go:209  [-] 482 +  | 	pkg/sql/internal.go:584
I250910 14:26:03.304650 50 kv/kvserver_test/client_spanconfig_backpressure_test.go:209  [-] 482 +  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).execIEStmt
I250910 14:26:03.304650 50 kv/kvserver_test/client_spanconfig_backpressure_test.go:209  [-] 482 +  | 	pkg/sql/internal.go:872
I250910 14:26:03.304650 50 kv/kvserver_test/client_spanconfig_backpressure_test.go:209  [-] 482 +  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).ExecEx
I250910 14:26:03.304650 50 kv/kvserver_test/client_spanconfig_backpressure_test.go:209  [-] 482 +  | 	pkg/sql/internal.go:820
.
.
.
```

**Commit 2: Implement the Fix**
- **Enhanced PTS Detection Logic** (`kvaccessor.go`): Implemented logic in `diffSpanConfigChangesForGCPolicyChangesOrPTSDeletes` to detect:
  - GCTTL changes (immediate bypass)
  - PTS deletions
  - Handles edge cases where PTS count remains the same but unique timestamps are removed
- **Transaction Name Labeling**: When GCTTL changes or PTS deletions are detected, the transaction is labeled with `"spanconfig-background-update"`
- **Backpressure Exemption** (`replica_backpressure.go`): Added transaction name checking in `canBackpressureBatch` to exempt transactions labeled as `"spanconfig-background-update"` from backpressure

##### Testing

The test is updated to show that the spanconfig updates have an exemption for GCTTL changes or PTS deletions.

The test validates backpressure bypass behaviour for various spanconfig update scenarios:
**Should Succeed (Bypass Backpressure):**
1. **Delete One PTS**: Removing a single protected timestamp from existing policies
2. **Remove All PTS**: Clearing all protected timestamps from the spanconfig
3. **Change GCTTL with No PTS**: Modifying garbage collection TTL to 0 with no protected timestamps

**Should Fail (Be Backpressured):**
4. **Add Extra PTS**: Adding additional protected timestamps should NOT bypass backpressure


Fixes: #146982
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/152515)
<!-- Reviewable:end -->

cockroach-teamcity commented:
<details><summary><strong>🔴 Sysbench</strong> [SQL, 3node, oltp_read_write]</summary>

| Metric                      | Old Commit     | New Commit     | Delta      | Note         |
|-----------------------------|----------------|----------------|------------|--------------|
| 🔴 **sec/op** | 7.912m ±0% | 8.920m ±0% | +12.74% | p=0.000 n=15    |
| 🔴 **allocs/op** | 7.817k ±0% | 12.30k ±0% | +57.30% | p=0.000 n=15    |

<details><summary>Reproduce</summary>

**benchdiff binaries**:
```shell
mkdir -p benchdiff/766a77e/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/766a77ea1b0726fc204eb2a095920cad739294cc/bin/pkg_sql_tests benchdiff/766a77e/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/766a77e/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
mkdir -p benchdiff/7b270fe/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/7b270fe87d233c4a21a469a6597f20d8f257c571/bin/pkg_sql_tests benchdiff/7b270fe/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/7b270fe/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
```
**benchdiff command**:
```shell
benchdiff --run=^BenchmarkSysbench/SQL/3node/oltp_read_write$ --old=7b270fe --new=766a77e ./pkg/sql/tests
```

</details>

</details>
<details><summary><strong>🔴 Sysbench</strong> [KV, 3node, oltp_read_only]</summary>

| Metric                      | Old Commit     | New Commit     | Delta      | Note         |
|-----------------------------|----------------|----------------|------------|--------------|
| 🔴 **sec/op** | 1.090m ±1% | 1.548m ±1% | +41.96% | p=0.000 n=15    |
| 🔴 **allocs/op** | 1.867k ±0% | 3.442k ±2% | +84.36% | p=0.000 n=15    |

<details><summary>Reproduce</summary>

**benchdiff binaries**:
```shell
mkdir -p benchdiff/766a77e/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/766a77ea1b0726fc204eb2a095920cad739294cc/bin/pkg_sql_tests benchdiff/766a77e/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/766a77e/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
mkdir -p benchdiff/7b270fe/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/7b270fe87d233c4a21a469a6597f20d8f257c571/bin/pkg_sql_tests benchdiff/7b270fe/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/7b270fe/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
```
**benchdiff command**:
```shell
benchdiff --run=^BenchmarkSysbench/KV/3node/oltp_read_only$ --old=7b270fe --new=766a77e ./pkg/sql/tests
```

</details>

</details>
<details><summary><strong>🔴 Sysbench</strong> [KV, 3node, oltp_write_only]</summary>

| Metric                      | Old Commit     | New Commit     | Delta      | Note         |
|-----------------------------|----------------|----------------|------------|--------------|
| 🔴 **sec/op** | 1.890m ±0% | 2.386m ±1% | +26.20% | p=0.000 n=15    |
| 🔴 **allocs/op** | 3.790k ±0% | 8.642k ±0% | +128.02% | p=0.000 n=15    |

<details><summary>Reproduce</summary>

**benchdiff binaries**:
```shell
mkdir -p benchdiff/766a77e/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/766a77ea1b0726fc204eb2a095920cad739294cc/bin/pkg_sql_tests benchdiff/766a77e/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/766a77e/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
mkdir -p benchdiff/7b270fe/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/7b270fe87d233c4a21a469a6597f20d8f257c571/bin/pkg_sql_tests benchdiff/7b270fe/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/7b270fe/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
```
**benchdiff command**:
```shell
benchdiff --run=^BenchmarkSysbench/KV/3node/oltp_write_only$ --old=7b270fe --new=766a77e ./pkg/sql/tests
```

</details>

</details>

<details><summary>Artifacts</summary>

**download**:
```shell
mkdir -p new
gcloud storage cp gs://cockroach-microbench-ci/artifacts/766a77ea1b0726fc204eb2a095920cad739294cc/17240696398-1/\* new/
mkdir -p old
gcloud storage cp gs://cockroach-microbench-ci/artifacts/7b270fe87d233c4a21a469a6597f20d8f257c571/17240696398-1/\* old/
```

</details>

_built with commit: [766a77ea1b0726fc204eb2a095920cad739294cc](https://github.com/cockroachdb/cockroach/commit/766a77ea1b0726fc204eb2a095920cad739294cc)_

<!-- microbench-ci -->

Commit:
kvserver: reproduce range-size backpressure preventing spanconfig updates

Previously, spanconfig updates could be blocked by range-size backpressure,
creating a catch-22 situation where garbage collection and protected timestamp
removal operations were prevented from running, which in turn prevented range
splitting and alleviation of the backpressure condition.

This was inadequate because it could lead to cluster-wide issues where
protected timestamp cleanup operations would fail indefinitely, preventing
backups, changefeeds, and other operations from completing properly.

To address this, this commit adds a comprehensive test that reproduces the
issue by repeatedly writing spanconfig updates for a single key until the
range becomes too large to split, triggering backpressure. The test
demonstrates how spanconfig updates (including protected timestamp deletions)
get blocked by backpressure, preventing the very operations needed to resolve
the backpressure condition.

Fixes: #146982
Release note: None

cockroach-teamcity commented:
<details><summary><strong>⚪ Sysbench</strong> [SQL, 3node, oltp_read_write]</summary>

| Metric                      | Old Commit     | New Commit     | Delta      | Note         |
|-----------------------------|----------------|----------------|------------|--------------|
| ⚪ **sec/op** | 7.854m ±0% | 7.864m ±0% | ~ | p=0.775 n=15    |
| ⚪ **allocs/op** | 7.816k ±2% | 7.825k ±2% | ~ | p=0.519 n=15    |

<details><summary>Reproduce</summary>

**benchdiff binaries**:
```shell
mkdir -p benchdiff/ce98286/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/ce98286ffd926b650a4a290165d8b122dc24bf11/bin/pkg_sql_tests benchdiff/ce98286/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/ce98286/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
mkdir -p benchdiff/47aa793/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/47aa7939ef2d5a783fc81c409f4454f1d65935ee/bin/pkg_sql_tests benchdiff/47aa793/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/47aa793/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
```
**benchdiff command**:
```shell
benchdiff --run=^BenchmarkSysbench/SQL/3node/oltp_read_write$ --old=47aa793 --new=ce98286 ./pkg/sql/tests
```

</details>

</details>
<details><summary><strong>⚪ Sysbench</strong> [KV, 3node, oltp_read_only]</summary>

| Metric                      | Old Commit     | New Commit     | Delta      | Note         |
|-----------------------------|----------------|----------------|------------|--------------|
| ⚪ **sec/op** | 1.095m ±1% | 1.098m ±1% | ~ | p=0.902 n=15    |
| ⚪ **allocs/op** | 1.868k ±0% | 1.868k ±0% | ~ | p=0.886 n=15    |

<details><summary>Reproduce</summary>

**benchdiff binaries**:
```shell
mkdir -p benchdiff/ce98286/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/ce98286ffd926b650a4a290165d8b122dc24bf11/bin/pkg_sql_tests benchdiff/ce98286/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/ce98286/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
mkdir -p benchdiff/47aa793/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/47aa7939ef2d5a783fc81c409f4454f1d65935ee/bin/pkg_sql_tests benchdiff/47aa793/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/47aa793/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
```
**benchdiff command**:
```shell
benchdiff --run=^BenchmarkSysbench/KV/3node/oltp_read_only$ --old=47aa793 --new=ce98286 ./pkg/sql/tests
```

</details>

</details>
<details><summary><strong>⚪ Sysbench</strong> [KV, 3node, oltp_write_only]</summary>

| Metric                      | Old Commit     | New Commit     | Delta      | Note         |
|-----------------------------|----------------|----------------|------------|--------------|
| ⚪ **sec/op** | 1.921m ±0% | 1.932m ±0% | ~ | p=0.041 n=15    |
| ⚪ **allocs/op** | 3.789k ±0% | 3.789k ±0% | ~ | p=0.451 n=15    |

<details><summary>Reproduce</summary>

**benchdiff binaries**:
```shell
mkdir -p benchdiff/ce98286/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/ce98286ffd926b650a4a290165d8b122dc24bf11/bin/pkg_sql_tests benchdiff/ce98286/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/ce98286/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
mkdir -p benchdiff/47aa793/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/47aa7939ef2d5a783fc81c409f4454f1d65935ee/bin/pkg_sql_tests benchdiff/47aa793/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/47aa793/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
```
**benchdiff command**:
```shell
benchdiff --run=^BenchmarkSysbench/KV/3node/oltp_write_only$ --old=47aa793 --new=ce98286 ./pkg/sql/tests
```

</details>

</details>

<details><summary>Artifacts</summary>

**download**:
```shell
mkdir -p new
gcloud storage cp gs://cockroach-microbench-ci/artifacts/ce98286ffd926b650a4a290165d8b122dc24bf11/17617843184-1/\* new/
mkdir -p old
gcloud storage cp gs://cockroach-microbench-ci/artifacts/47aa7939ef2d5a783fc81c409f4454f1d65935ee/17617843184-1/\* old/
```

</details>

_built with commit: [ce98286ffd926b650a4a290165d8b122dc24bf11](https://github.com/cockroachdb/cockroach/commit/ce98286ffd926b650a4a290165d8b122dc24bf11)_

<!-- microbench-ci -->

Commit:
kvserver: fix spanconfig backpressure bypass for GCTTL/PTS updates

Previously, spanconfig updates involving garbage collection TTL changes
or protected timestamp deletions could be blocked by range-size backpressure,
creating a catch-22 situation where cleanup operations were prevented from
running, which in turn prevented range splitting and alleviation of the
backpressure condition.

This was inadequate because it could lead to cluster-wide issues where
protected timestamp cleanup operations would fail indefinitely, preventing
backups, changefeeds, and other operations from completing properly.

To address this, this commit:
- Implements sophisticated logic in `kvaccessor.go` to detect GCTTL changes
  and PTS deletions by comparing existing and new spanconfigs, including
  handling of duplicate timestamps and net deletions
- Adds transaction name checking in `replica_backpressure.go` to exempt
  transactions labeled as "spanconfig-background-update" from backpressure
- Adds comprehensive test coverage in `client_spanconfig_backpressure_test.go`
  to validate backpressure bypass behavior for various spanconfig update
  scenarios (GCTTL changes, PTS deletions, and PTS additions)

The fix ensures that spanconfig updates involving GCTTL changes or PTS
deletions are properly labeled with the "spanconfig-background-update"
transaction name, allowing them to bypass backpressure and prevent
the catch-22 situation.

Fixes: #146982
Release note: None

cockroach-teamcity commented:
<details><summary><strong>⚪ Sysbench</strong> [SQL, 3node, oltp_read_write]</summary>

| Metric                      | Old Commit     | New Commit     | Delta      | Note         |
|-----------------------------|----------------|----------------|------------|--------------|
| ⚪ **sec/op** | 7.952m ±1% | 7.950m ±1% | ~ | p=0.838 n=15    |
| ⚪ **allocs/op** | 7.819k ±2% | 7.809k ±2% | ~ | p=0.690 n=15    |

<details><summary>Reproduce</summary>

**benchdiff binaries**:
```shell
mkdir -p benchdiff/3fbe842/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/3fbe842f0bb88fb52590831b462092f69dd2634d/bin/pkg_sql_tests benchdiff/3fbe842/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/3fbe842/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
mkdir -p benchdiff/47aa793/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/47aa7939ef2d5a783fc81c409f4454f1d65935ee/bin/pkg_sql_tests benchdiff/47aa793/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/47aa793/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
```
**benchdiff command**:
```shell
benchdiff --run=^BenchmarkSysbench/SQL/3node/oltp_read_write$ --old=47aa793 --new=3fbe842 ./pkg/sql/tests
```

</details>

</details>
<details><summary><strong>⚪ Sysbench</strong> [KV, 3node, oltp_read_only]</summary>

| Metric                      | Old Commit     | New Commit     | Delta      | Note         |
|-----------------------------|----------------|----------------|------------|--------------|
| ⚪ **sec/op** | 1.085m ±1% | 1.087m ±2% | ~ | p=0.305 n=15    |
| ⚪ **allocs/op** | 1.868k ±0% | 1.867k ±0% | ~ | p=0.440 n=15    |

<details><summary>Reproduce</summary>

**benchdiff binaries**:
```shell
mkdir -p benchdiff/3fbe842/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/3fbe842f0bb88fb52590831b462092f69dd2634d/bin/pkg_sql_tests benchdiff/3fbe842/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/3fbe842/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
mkdir -p benchdiff/47aa793/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/47aa7939ef2d5a783fc81c409f4454f1d65935ee/bin/pkg_sql_tests benchdiff/47aa793/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/47aa793/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
```
**benchdiff command**:
```shell
benchdiff --run=^BenchmarkSysbench/KV/3node/oltp_read_only$ --old=47aa793 --new=3fbe842 ./pkg/sql/tests
```

</details>

</details>
<details><summary><strong>⚪ Sysbench</strong> [KV, 3node, oltp_write_only]</summary>

| Metric                      | Old Commit     | New Commit     | Delta      | Note         |
|-----------------------------|----------------|----------------|------------|--------------|
| ⚪ **sec/op** | 1.908m ±1% | 1.912m ±0% | ~ | p=0.838 n=15    |
| ⚪ **allocs/op** | 3.787k ±0% | 3.788k ±0% | ~ | p=0.673 n=15    |

<details><summary>Reproduce</summary>

**benchdiff binaries**:
```shell
mkdir -p benchdiff/3fbe842/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/3fbe842f0bb88fb52590831b462092f69dd2634d/bin/pkg_sql_tests benchdiff/3fbe842/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/3fbe842/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
mkdir -p benchdiff/47aa793/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/47aa7939ef2d5a783fc81c409f4454f1d65935ee/bin/pkg_sql_tests benchdiff/47aa793/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/47aa793/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
```
**benchdiff command**:
```shell
benchdiff --run=^BenchmarkSysbench/KV/3node/oltp_write_only$ --old=47aa793 --new=3fbe842 ./pkg/sql/tests
```

</details>

</details>

<details><summary>Artifacts</summary>

**download**:
```shell
mkdir -p new
gcloud storage cp gs://cockroach-microbench-ci/artifacts/3fbe842f0bb88fb52590831b462092f69dd2634d/17621706729-1/\* new/
mkdir -p old
gcloud storage cp gs://cockroach-microbench-ci/artifacts/47aa7939ef2d5a783fc81c409f4454f1d65935ee/17621706729-1/\* old/
```

</details>

_built with commit: [3fbe842f0bb88fb52590831b462092f69dd2634d](https://github.com/cockroachdb/cockroach/commit/3fbe842f0bb88fb52590831b462092f69dd2634d)_

<!-- microbench-ci -->

cockroach-teamcity commented:
<details><summary><strong>⚪ Sysbench</strong> [SQL, 3node, oltp_read_write]</summary>

| Metric                      | Old Commit     | New Commit     | Delta      | Note         |
|-----------------------------|----------------|----------------|------------|--------------|
| ⚪ **sec/op** | 7.961m ±0% | 7.960m ±1% | ~ | p=0.775 n=15    |
| ⚪ **allocs/op** | 7.817k ±2% | 7.814k ±0% | ~ | p=0.829 n=15    |

<details><summary>Reproduce</summary>

**benchdiff binaries**:
```shell
mkdir -p benchdiff/7264157/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/726415749b89cce103857040bf56adf6f3291942/bin/pkg_sql_tests benchdiff/7264157/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/7264157/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
mkdir -p benchdiff/95e1d21/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/95e1d211b786f1791afe3c9f9276fa6aa0032bd8/bin/pkg_sql_tests benchdiff/95e1d21/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/95e1d21/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
```
**benchdiff command**:
```shell
benchdiff --run=^BenchmarkSysbench/SQL/3node/oltp_read_write$ --old=95e1d21 --new=7264157 ./pkg/sql/tests
```

</details>

</details>
<details><summary><strong>⚪ Sysbench</strong> [KV, 3node, oltp_read_only]</summary>

| Metric                      | Old Commit     | New Commit     | Delta      | Note         |
|-----------------------------|----------------|----------------|------------|--------------|
| ⚪ **sec/op** | 1.096m ±2% | 1.088m ±1% | ~ | p=0.512 n=15    |
| ⚪ **allocs/op** | 1.868k ±0% | 1.868k ±0% | ~ | p=0.942 n=15    |

<details><summary>Reproduce</summary>

**benchdiff binaries**:
```shell
mkdir -p benchdiff/7264157/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/726415749b89cce103857040bf56adf6f3291942/bin/pkg_sql_tests benchdiff/7264157/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/7264157/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
mkdir -p benchdiff/95e1d21/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/95e1d211b786f1791afe3c9f9276fa6aa0032bd8/bin/pkg_sql_tests benchdiff/95e1d21/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/95e1d21/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
```
**benchdiff command**:
```shell
benchdiff --run=^BenchmarkSysbench/KV/3node/oltp_read_only$ --old=95e1d21 --new=7264157 ./pkg/sql/tests
```

</details>

</details>
<details><summary><strong>⚪ Sysbench</strong> [KV, 3node, oltp_write_only]</summary>

| Metric                      | Old Commit     | New Commit     | Delta      | Note         |
|-----------------------------|----------------|----------------|------------|--------------|
| ⚪ **sec/op** | 1.923m ±1% | 1.918m ±0% | ~ | p=0.056 n=15    |
| ⚪ **allocs/op** | 3.793k ±0% | 3.792k ±0% | ~ | p=0.658 n=15    |

<details><summary>Reproduce</summary>

**benchdiff binaries**:
```shell
mkdir -p benchdiff/7264157/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/726415749b89cce103857040bf56adf6f3291942/bin/pkg_sql_tests benchdiff/7264157/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/7264157/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
mkdir -p benchdiff/95e1d21/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/95e1d211b786f1791afe3c9f9276fa6aa0032bd8/bin/pkg_sql_tests benchdiff/95e1d21/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/95e1d21/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
```
**benchdiff command**:
```shell
benchdiff --run=^BenchmarkSysbench/KV/3node/oltp_write_only$ --old=95e1d21 --new=7264157 ./pkg/sql/tests
```

</details>

</details>

<details><summary>Artifacts</summary>

**download**:
```shell
mkdir -p new
gcloud storage cp gs://cockroach-microbench-ci/artifacts/726415749b89cce103857040bf56adf6f3291942/17624183743-1/\* new/
mkdir -p old
gcloud storage cp gs://cockroach-microbench-ci/artifacts/95e1d211b786f1791afe3c9f9276fa6aa0032bd8/17624183743-1/\* old/
```

</details>

_built with commit: [726415749b89cce103857040bf56adf6f3291942](https://github.com/cockroachdb/cockroach/commit/726415749b89cce103857040bf56adf6f3291942)_

<!-- microbench-ci -->
